docs/design-proposals/2022-04 TraceQL Concepts.md

```diff
@@ -0,0 +1,198 @@
+`{ region = "eu-west-0" } && { region = "eu-west-1" } `
+
+Note the difference between the above and the following:  
+`{ region = "eu-west-0"  && region = "eu-west-1" } `
+
+The second expression will return no traces because it's impossible for both conditions to be simultaneously true on the same span.
+
+### Structural operators
+
+Structural operators evaluate traces based on the span relationships within the span tree. 
+
+`{ } >> { }`  
+  This is the descendant operator. The spans returned from this operator will match the right hand side conditions while also being descendants of spans that match the left hand side conditions.
+
+`{ } > { }`  
+  This is the child operator. The spans returned from this operator will match the right hand side conditions while also being children of spans that match the left hand side conditions.
+
+`{ } ~ { }`  
+  This is the sibling operator. The spans returned from this operator will match the right hand side conditions while also being siblings of spans that match the left hand side conditions.
+
+## Aggregators
+
+All of the above expressions involve asking questions about individual spans. However, sometimes we want to ask questions about a set of spans. For that we have aggregate functions.
+
+**Examples**
+
+Find traces where the total number of spans is greater than 10:  
+`count() > 10`
+
+Find traces where the average duration is greater than 1s:  
+`avg(duration) > 1s`
+
+## Expression Pipelining
+
+Pipelining allows us to "pipe" a set of spans from one expression to the next. This is particularly useful if you want to perform an aggregate over a subset of a trace.
+
+**Examples**
+
+Find traces where that have more than 3 spans with an attribute`http.status` with a value of `200`:  
+`{ http.status = 200 } | count() > 3`
+
+## Grouping
+
+Grouping allows us to take a trace and break it down into sets of spans that are evaluated by future pipeline entries. Each set of spans created by the group is _individually_ evaluated by downstream expressions.
+
+**Examples**
+
+Find traces that have more than 5 spans in any region:  
+`by(region) | count() > 5 `
+
+## Examples
+
+Any span matches an attribute:  
+`{ namespace = "prod" }`
+
+Two attributes appear on the same span:  
+`{ namespace = "prod" && http.status = 200 }`
+
+Two attributes appear anywhere within a trace:  
+`{ namespace = "prod" } && { http.status = 200 }`
+
+Any span has a duration over one second:  
+`{ duration > 1s }`
+
+The trace as a whole has a duration of over one second:  
+`max(end) - min(start) > 1s`
+
+Any span has a duration of over one second and any span has a specific attribute. These spans can be the same or different:  
+`max(duration) > 1s && { namespace = "prod" }`
+
+A trace has over 5 spans with http.status = 200 in any given namespace:  
+`by(namespace) | { http.status = 200 } | count() > 5`
```

Comment:
```suggestion
`{ http.status = 200 } | by(namespace) | count() > 5`
```
How about switching the order - reduce then group.  This seems like it should be more efficient, but also is clearer conceptually.